File: exporter-cli.js
#!/usr/bin/env node

const readline = require('readline');
const AdvancedTablePDFExporter = require('./table-pdf-exporter-advanced');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function question(prompt) {
  return new Promise(resolve => {
    rl.question(prompt, resolve);
  });
}

async function main() {
  console.clear();
  console.log('╔════════════════════════════════════════════════════════╗');
  console.log('║        🌟 جهاز تصدير جداول منصة STAR1 الأوتوماتيكي 🌟 ║');
  console.log('╚════════════════════════════════════════════════════════╝\n');

  // Get credentials
  const username = await question('👤 اسم المستخدم: ');
  const password = await question('🔑 كلمة المرور: ');

  // Get decision registration numbers
  console.log('\n📋 أدخل أرقام قرارات التسجيل (واحد في السطر)');
  console.log('   (اضغط Enter مرتين للانتهاء)\n');

  const decisions = [];
  let decisionNumber = await question('رقم قرار التسجيل #1: ');

  while (decisionNumber.trim()) {
    decisions.push({
      decisionNumber: decisionNumber.trim(),
      name: `قرار_${decisionNumber.trim()}`
    });
    decisionNumber = await question(`رقم قرار التسجيل #${decisions.length + 1}: `);
  }

  if (decisions.length === 0) {
    console.log('❌ لم تدخل أي قرارات!');
    rl.close();
    return;
  }

  console.log(`\n✅ سيتم تصدير ${decisions.length} قرار\n`);

  // Ask for confirmation
  const confirm = await question('هل تريد المتابعة؟ (نعم/لا): ');

  if (!confirm.toLowerCase().startsWith('ن')) {
    console.log('❌ تم الإلغاء');
    rl.close();
    return;
  }

  rl.close();

  // Start export
  console.log('\n🚀 جارٍ البدء بالتصدير...\n');

  const exporter = new AdvancedTablePDFExporter(username, password, {
    headless: true,
    outputDir: './exports',
  });

  try {
    const results = await exporter.run(decisions);

    console.log('\n╔════════════════════════════════════════════════════════╗');
    console.log('║                  النتائج النهائية                    ║');
    console.log('╚════════════════════════════════════════════════════════╝\n');

    results.forEach((r, i) => {
      if (r.status === 'success') {
        console.log(`✅ [${i + 1}] القرار ${r.decisionNumber}`);
        console.log(`   📄 PDF: ${r.pdfPath}`);
        console.log(`   📸 لقطة: ${r.pngPath}\n`);
      } else {
        console.log(`❌ [${i + 1}] القرار ${r.decisionNumber}`);
        console.log(`   الخطأ: ${r.error}\n`);
      }
    });

    const successCount = results.filter(r => r.status === 'success').length;
    console.log(`📊 النتيجة النهائية: ${successCount}/${results.length}`);

    if (successCount === results.length) {
      console.log('🎉 تم تصدير جميع القرارات بنجاح!');
    }

    if (successCount === results.length) {
      console.log('🎉 تم التصدير بنجاح!');
    }
  } catch (error) {
    console.error('\n❌ خطأ:', error.message);
    process.exit(1);
  }
}

main();

File: exporter_cli.py
#!/usr/bin/env python3

import asyncio
import sys
from getpass import getpass
from table_pdf_exporter_advanced import AdvancedTablePDFExporter


async def main():
    """Interactive CLI for the exporter"""
    print('\n╔════════════════════════════════════════════╗')
    print('║  🚀 أداة تصدير جداول معايير التصنيف - PDF   ║')
    print('╚════════════════════════════════════════════╝\n')

    # Get credentials
    username = input('👤 اسم المستخدم: ').strip()
    password = getpass('🔑 كلمة المرور: ')

    if not username or not password:
        print('❌ يجب إدخال اسم المستخدم وكلمة المرور')
        sys.exit(1)

    # Get decision numbers
    print('\n📋 أدخل أرقام قرارات التسجيل (واحد في السطر):')
    print('   (اضغط Enter مرتين للانتهاء)\n')

    decision_numbers = []
    counter = 1

    while True:
        decision = input(f'رقم قرار التسجيل #{counter}: ').strip()

        if not decision:
            if decision_numbers:
                break
            print('   ⚠️ يجب إدخال رقم قرار واحد على الأقل')
            continue

        decision_numbers.append(decision)
        counter += 1

    # Confirm
    print(f'\n✅ سيتم تصدير {len(decision_numbers)} قرار')
    confirm = input('هل تريد المتابعة؟ (نعم/لا): ').strip().lower()

    if confirm not in ['نعم', 'yes', 'y']:
        print('❌ تم الإلغاء')
        sys.exit(1)

    # Prepare decisions
    decisions = [
        {'name': f'قرار_{i + 1}', 'decisionNumber': num}
        for i, num in enumerate(decision_numbers)
    ]

    # Run exporter
    print('\n' + '=' * 44)
    exporter = AdvancedTablePDFExporter(username, password)

    try:
        results = await exporter.run(decisions)

        print('\n╔════════════════════════════════════════╗')
        print('║             النتائج النهائية           ║')
        print('╚════════════════════════════════════════╝')

        for r in results:
            if r['status'] == 'success':
                print(f"✅ {r.get('siteName') or r['name']}: {r['pdfPath']}")
            else:
                print(f"❌ {r['name']}: {r['error']}")

        success_count = sum(1 for r in results if r['status'] == 'success')
        print(f'\n📊 النتيجة: {success_count}/{len(results)}')

        if success_count == len(results):
            print('\n🎉 تم التصدير بنجاح!')
        else:
            print(f'\n⚠️ تم تصدير {success_count} من {len(results)} قرار')

    except Exception as e:
        print(f'\n❌ فشل التنفيذ: {str(e)}')
        sys.exit(1)


if __name__ == '__main__':
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        print('\n\n⚠️ تم الإيقاف بواسطة المستخدم')
        sys.exit(0)
